composer.json

```diff
@@ -31,7 +31,7 @@
         "league/csv": "^8 || ^9",
         "m1/env": "^2.1",
         "monolog/monolog": "^1.16",
-        "nikic/php-parser": "^3 || ^4",
+        "nikic/php-parser": "^3 || ^4.10.5",
```

Comment:
`^4.10.5` is technically only required if you’re using `enum`, with lower versions you’ll encounter [errors like this](https://github.com/silverstripe/silverstripe-framework/runs/7470656402?check_suite_focus=true#step:12:115) on dev/build.

There’s not really anything stopping you using versions prior to 4.10.5 as long as you’re not using enums, but our PHP 7.4 CI runs use `--prefer-lowest` when composer installing, which causes the error linked above.

The only other alternatives I can think of are:

1. Stop using `--prefer-lowest` for our CI runs. I assume there’s a reason this is in-place so I haven’t looked at this.
2. Somehow get our manifest file finder to _not_ pass enums to php-parser if running on PHP 8.1, or a version of php-parser <4.10.5. That just seems incredible messy.
3. Separate the enum files into their own test manifest (give them their own directory like `fixtures/classmanifest_enums`), build that manifest in each enum-related test. Shouldn’t be too hard, but then if we ever have `--prefer-lowest` on PHP 8.1 builds in future (or someone adds that flag themselves) we’ll get the same errors.

To be honest, supporting core language features shouldn’t be at the mercy of a `--prefer-lowest` composer flag so IMO it makes more sense to just bump the minimum version.

Should probably consider dropping 3.x support too.

Reply:
I see nothing wrong with bumping this constraint. It's unlikely to be required directly in projects.
But won't keeping the `3` part of this constraint still result in failures on `--prefer-lowest` CI builds? I guess not given that it hasn't failed on this PR but I'd like to understand why that isn't failing... if `3` actually just _isn't_ installable for some reason (and therefore not being used by the `--prefer-lowest` build) then we should just remove that part of the constraint.

Reply:
> But won't keeping the `3` part of this constraint still result in failures on `--prefer-lowest` CI builds? I guess not given that it hasn't failed on this PR but I'd like to understand why that isn't failing...

I think it’s because `nikic/php-parser: 3` requires `phpunit/phpunit: ~4.0|~5.0`, which doesn’t match our constraints. So I suppose we therefore can remove `3`? _Technically_ it could be possible to install 3 with `--no-dev` but having differing major package versions between dev/no-dev seems daft

Reply:
Yeah I agree. The whole point of `--prefer-lowest` is to be able to validate that our lowest constraints work. We can't do that without Dev dependencies so we can't test the `3` major there. 

All that being a long way of saying that I'd be inclined to remove `3` there.